src/main/scala/format/Forsyth.scala

```diff
@@ -50,43 +50,163 @@ object Forsyth {
     case _                                                 => sys.error("Mismatched gamelogic types 15")
   }
 
-  case class SituationPlus(situation: Situation, fullTurnCount: Int) {
+  sealed trait SituationPlus {
 
-    def turnCount = fullTurnCount * 2 - situation.player.fold(2, 1)
-    def plies     = turnCount
+    val situation: Situation
+    val fullTurnCount: Int
+    def turnCount: Int
+    def plies: Int
 
   }
 
+  // TODO: This is probably a situation where bog-standard inheritance works
```

Comment:
can probably undo all of this if we aren't needing to pass anything extra into `SituationPlus` because its all in `board.history`